### ISSUE TITLE
Bug: `js` files are linted even though `files` only includes `ts` and `tsx`

### ISSUE THREAD

**Issue by 0-CAT** [eslint/eslint#18975]
### Environment

Node version: v20.15.0
npm version: v10.7.0
Local ESLint version: v9.11.1 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.1.0 (macOS Sonoma)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
import globals from "globals";
import typescriptEslint from "typescript-eslint";

export default [
	{
		files: ["**/*.ts", "**/*.tsx"],
		languageOptions: {
			ecmaVersion: 5,
			globals: {
				...globals.node,
			},
			parser: typescriptEslint.parser,
			parserOptions: {
				project: "tsconfig.base.json",
			},
			sourceType: "module",
		},
		rules: {},
	},
];
```
</details>

File content (`my.js`):

```js
/* eslint-disable */
```

Minimal repro: [eslint-issue.zip](https://github.com/user-attachments/files/17219165/eslint-issue.zip)

Just run `npm i` and then `npm run lint`. I would expect there to be no issues, but instead, a warning about the unused `eslint-disable` rule is printed out:

![Image](https://github.com/user-attachments/assets/228f5677-bfc4-46cf-92b2-5f48123f77d2)

### What did you expect to happen?

Running `eslint` should lint only `.ts` and `.tsx` files.

### What actually happened?

Running `eslint` caused `.js` files to also be linted.

### Link to Minimal Reproducible Example

Linked in "What did you do?" section.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

- In the minimal repro, I specified no rules, but where I originally discovered this issue, I have many, many rules defined, so that's not related.
- This even happens if I specify `ignores: ["**/*.js"]` or `ignores: ["my.js"]` in the configuration.
- I was also having this issue in 8.57.1, but upgrading to 9.11.1 did not help.

I also thought this might be related to the parser, but `my.js` in my minimal repro gets linted even with the following configuration which does not use TS-ESLint:

```js
export default [
  {
    files: ["**/*.jsx"],
    ignores: ["my.js"],
    languageOptions: {
      ecmaVersion: 5,
      globals: {
        ...globals.node,
      },
      sourceType: "module",
    },
    rules: {},
  },
];
```

**Comment by 0-CAT** [comment]
One other comment: This only started after I migrated to flat config. I was previously using `--ext=ts,tsx` from the CLI which worked fine.

**Comment by mdjermanovic** [comment]
This works as intended because .js, .mjs, and .cjs files are linted by default.

If you want to exclude .js files from linting, you can use [global ignores](https://eslint.org/docs/latest/use/configure/ignore):

```js
import globals from "globals";
import typescriptEslint from "typescript-eslint";

export default [
	{
		files: ["**/*.ts", "**/*.tsx"],
		languageOptions: {
			ecmaVersion: 5,
			globals: {
				...globals.node,
			},
			parser: typescriptEslint.parser,
			parserOptions: {
				project: "tsconfig.base.json",
			},
			sourceType: "module",
		},
		rules: {},
	},
	{
		ignores: ["**/*.js"]
	},
];
```

**Comment by 0-CAT** [comment]
> This works as intended because .js, .mjs, and .cjs files are linted by default.
> 
> If you want to exclude .js files from linting, you can use [global ignores](https://eslint.org/docs/latest/use/configure/ignore):
>
> ``` js
> import globals from "globals";
> import typescriptEslint from "typescript-eslint";
> 
> export default [
> 	{
> 		files: ["**/*.ts", "**/*.tsx"],
> 		languageOptions: {
> 			ecmaVersion: 5,
> 			globals: {
> 				...globals.node,
> 			},
> 			parser: typescriptEslint.parser,
> 			parserOptions: {
> 				project: "tsconfig.base.json",
> 			},
> 			sourceType: "module",
> 		},
> 		rules: {},
> 	},
> 	{
> 		ignores: ["**/*.js"]
> 	},
> ];
> ```

Thanks for the response. I can confirm that your workaround works.

It may be worth documenting this, given that the doc on `files` and `ignores` does not mention that there are any "by default" values or values that otherwise will be linted if not explicitly excluded in the global ignores.

![Image](https://github.com/user-attachments/assets/39193bd4-7d06-4e2a-9cc7-87ebf40c1fec)

The reason that I ran into this was that because TS compiles to JS, effectively my entire project's minified output was being linted because I couldn't use my main configuration's `ignores` property to ignore it.

**Comment by mdjermanovic** [comment]
> It may be worth documenting this, given that the doc on `files` and `ignores` does not mention that there are any "by default" values or values that otherwise will be linted if not explicitly excluded in the global ignores.

It's mentioned here:

https://eslint.org/docs/latest/use/configure/configuration-files#specifying-files-and-ignores

![Image](https://github.com/user-attachments/assets/a2ee33ff-fa43-4ff1-b9fb-992d0152bd38)

I made PR https://github.com/eslint/eslint/pull/18976 to clarify that global ignores should be used.
